scriptworker_client/tests/test_github_client.py

```diff
@@ -214,3 +214,284 @@ async def test_get_branch_head_oid_branch_not_found(aioresponses, github_client)
 
     with pytest.raises(UnknownBranchError, match="branch 'branchy' not found in repo!"):
         await github_client.get_branch_head_oid("branchy")
+
+
+@pytest.mark.asyncio
+async def test_get_repository_files(aioresponses, github_client):
```

Comment:
Just because you asked for feedback on tests, this seems fine. It's definitely ugly, but I'm not really sure how else to do it.